Channel health check improvement for cancelled requests (#36225)
Azure/azure-sdk-for-java: sdk/cosmos/azure-cosmos/CHANGELOG.md

#### Bugs Fixed
* Fixed retrying write operations when a gone exception occurs in bulk mode. - See [PR 35838](https://github.com/Azure/azure-sdk-for-java/pull/35838)
* Fixed request start time in the `CosmosDiagnostics` for individual request responses - See [PR 35705](https://github.com/Azure/azure-sdk-for-java/pull/35705)
* Fixed an issue where `ConnectionStateListener` tracked staled `Uris` which fails to mark the current `Uris` unhealthy properly - See [PR 36067](https://github.com/Azure/azure-sdk-for-java/pull/36067)
* Gone exceptions that are not idempotent should not be retried because it is not known if they succeeded for sure. The handling of the exception in this case is left to the user. Fixed retrying write operations when a gone exception occurs in bulk mode. - See [PR 35838](https://github.com/Azure/azure-sdk-for-java/pull/35838)
* Fixed an issue to update the last unhealthy timestamp for an `Uri` instance only when transitioning to `Unhealthy` from a different health status -  See [36083](https://github.com/Azure/azure-sdk-for-java/pull/36083)

> * Improved the channel health check flow to deem a channel which sees cancellations as unhealthy. - See [PR 36225](https://github.com/Azure/azure-sdk-for-java/pull/36225)

Reviewer: ```suggestion
* Improved the channel health check flow to deem a channel unhealthy when seen consecutive cancellations. - See [PR 36225](https://github.com/Azure/azure-sdk-for-java/pull/36225)
```

Author: Fixed in next iteration.
